server/market/bookrouter.go

```diff
@@ -0,0 +1,389 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package market
+
+import (
+	"context"
+	"encoding/json"
+	"sync"
+
+	"github.com/decred/dcrdex/server/comms"
+	"github.com/decred/dcrdex/server/comms/msgjson"
+	"github.com/decred/dcrdex/server/market/types"
+	"github.com/decred/dcrdex/server/order"
+)
+
+// An update action classifies updates into how they affect the book or epoch
+// queue.
+type updateAction uint8
+
+const (
+	// epochAction means an order is being added to the epoch queue and will
+	// result in a msgjson.EpochOrderNote being sent to subscribers.
+	epochAction updateAction = iota
+	// bookAction means an order is being added to the order book, and will result
+	// in a msgjson.BookOrderNote being sent to subscribers.
+	bookAction
+	// unbookAction means an order is being removed from the order book and will
+	// result in a msgjson.UnbookOrderNote being sent to subscribers.
+	unbookAction
+)
+
+// updateSignal combines an updateAction with the order on which the action
+// applies.
+type updateSignal struct {
+	action updateAction
+	order  order.Order
+}
+
+// BookSource is a source of a market's order book, as well as a feed of updates
+// to the order book and epoch queue.
+type BookSource interface {
+	Book() (buys []*order.LimitOrder, sells []*order.LimitOrder)
+	OrderFeed() chan *updateSignal
+}
```

Comment:
Right now, there is no syncing of the epoch queue on new connections, which means that the client cannot independently verify the matching for the epoch in which they connect, only for subsequent epochs for which they were connected for the full duration.